src/ByteSizeLib/ByteSize.cs

```diff
@@ -305,9 +305,43 @@ public ByteSize Subtract(ByteSize bs)
 
         public static ByteSize Parse(string s)
         {
+            return TryParse(s).Value;
+        }
+
+        public static bool TryParse(string s, out ByteSize result)
+        {
+            var parseResult = TryParse(s, CachedError);
+            result = parseResult.GetValueOrDefault();
+            return !parseResult.IsError;
+        }
+
+        static readonly Exception CachedError = new Exception();
```

Comment:
Are there potential thread safety issues with this given it's a static property?